components/camel-restlet/src/test/java/org/apache/camel/component/restlet/RestRestletCorsTest.java

```diff
@@ -0,0 +1,73 @@
+package org.apache.camel.component.restlet;
+
+import java.util.Set;
+
+import com.google.common.base.Splitter;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.RouteBuilder;
+import org.junit.Test;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static org.hamcrest.CoreMatchers.is;
+
+public class RestRestletCorsTest extends RestletTestSupport {
+    Splitter headerSplitter = Splitter.on(",").trimResults();
+
+    @Test
+    public void testRestletProducerGet() throws Exception {
+        Exchange exchange = template.request("http://localhost:" + portNum + "/users/123/basic", null);
+
+        // verify no problems have occurred:
+        assertFalse(exchange.isFailed());
+        Message message = exchange.getOut();
+        assertThat(message.getHeader(Exchange.HTTP_RESPONSE_CODE, Integer.class), is(200));
+
+        // verify all header values match those specified in restConfiguration:
+        assertThat(message.getHeader("Access-Control-Allow-Origin", String.class), is("https://localhost:443"));
```

Comment:
For some reason, Rest DSL supports only these 4 CORS headers. Although `DefaultRestletBinding` was updated to support more headers, those extra headers cannot be set via `.corsHeaderProperty()`.